vicodyn/include/cocaine/vicodyn/request_context.hpp

```diff
@@ -38,11 +40,7 @@ class request_context_t: public std::enable_shared_from_this<request_context_t>
 
     auto peer_use_count(const std::string& peer_uuid) -> size_t;
 
-    auto register_retry() -> void;
-
-    auto retry_count() -> size_t;
-
-    auto custom_context() -> synchronized<boost::any>&;
+    auto counter(const std::string& name) -> std::atomic<std::size_t>&;
```

Comment:
It is not so important, but you are exposing internal realization details to class caller, after receiving the reference, caller would in general know that it is an atomic and could start treat it as specifically as atomic in caller code.

Reply:
Yes, I'm agree. But usually a check of the counter and an increment of it are used very close, in the scope of one function. Current interface allows to use reference to a value and do the less count of getting counter by key. As a variant I can propose to return special class with method like "get" and method like "increment", but it seems like an overkill.